resource/resource.go

```diff
@@ -111,16 +111,16 @@ func (r *Resource) ParentField() string {
 	return r.parentField
 }
 
-// Compile the resource graph and report any error
-func (r *Resource) Compile() error {
+// Compile the resource graph and report any error.
+func (r *Resource) Compile(rl schema.ReferenceChecker) error {
```

Comment:
sholud be rc for consistency. the rl name is leftover from when it was called ReferenceLookup